layout/AddressesUsingIdsQuery.js

```diff
@@ -104,14 +163,34 @@ function createStreetShould(vs) {
     should.bool.boost = vs.var('boost:street');
   }
 
-  return should;
+  // Give a little should boost to features that have a parent id that
+  // perfectly matches part of the searched for admin hierarchy,
+  // so they beat out hits that were retrieved based on bounding box containment
+  const layer_filters = createLayerIdsShould(vs);
+  if (!_.isEmpty(layer_filters)) {
+    should.bool.should = [...should.bool.should || [], layer_filters];
+  }
 
+  return should;
 }
 
-function createLayerIdsShould(layer, ids) {
-  // create an object initialize with terms.'parent.locality_id' (or whatever)
-  // must use array syntax for 2nd parameter as _.set interprets '.' as new object
-  return _.set({}, ['terms', `parent.${layer}_id`], ids);
+function createLayerBoundingBoxesShould(vs, bboxes, scale = 1.0) {
+  return bboxes.map((bbox) => {
+    if (bbox.min_lat === bbox.max_lat || bbox.min_lon === bbox.max_lon) {
+      return;
+    }
+
+    var poly = turf.bboxPolygon([bbox.min_lon, bbox.min_lat, bbox.max_lon, bbox.max_lat]);
+    var scaledPoly = turf.transformScale(poly, scale);
+    const [minX, minY, maxX, maxY]  = turf.bbox(scaledPoly);
+
+    return _.set({}, `geo_bounding_box.${vs.var('centroid:field').$}`, {
+      'top': maxY,
+      'right': maxX,
+      'bottom': minY,
+      'left': minX,
+    });
+  }).filter(Boolean);
```

Comment:
What is the function of `.filter(Boolean)` here?

Reply:
it filters out undefineds from the early return in the map function b/c js doesn't have a good flatMap.

changed to   }).filter(clause => clause !== undefined)